packages/devtools_app/lib/src/screens/network/offline_network_data.dart

```diff
@@ -15,68 +14,49 @@ import 'network_model.dart';
 /// It is responsible for serializing and deserializing offline network data.
 class OfflineNetworkData with Serializable {
   OfflineNetworkData({
-    required this.requests,
+    required this.httpRequestData,
     this.selectedRequestId,
-    required this.currentRequests,
-    required this.socketStats,
+    required this.socketData,
   });
 
   /// Creates an instance of [OfflineNetworkData] from a JSON map.
   factory OfflineNetworkData.fromJson(Map<String, dynamic> json) {
-    final List<dynamic> requestsJson = json['requests'] ?? [];
-    final List<HttpProfileRequest>? currentReqData = json['currentRequests'];
-    final List<SocketStatistic>? socketStats = json['socketStats'];
-
-    final requests = requestsJson
-        .map(
-          (e) => DartIOHttpRequestData.fromJson(
-            e as Map<String, dynamic>,
-            null,
-            null,
-          ),
-        )
-        .toList();
+    final httpRequestData = json[OfflineDataKeys.httpRequestData.name]
+        as List<DartIOHttpRequestData>;
```

Comment:
cast this to the proper type instead:
`final httpRequestData = (json[OfflineDataKeys.httpRequestData.name] as List?)?.cast<DartIOHttpRequestData>();`

Reply:
now I have explicitly called fromJson on each object from list, will refactor it and make it into an extension.
BTW one observation is when I remove/comment all the toJson() of all the required classes, and pass the list of DartIOHttpRequestData directly 

```dart
//To Json
  Map<String, dynamic> toJson() {
    return {
      OfflineDataKeys.httpRequestData.name: httpRequestData,
      OfflineDataKeys.selectedRequestId.name: selectedRequestId,
      OfflineDataKeys.socketData.name: socketData,
    };
// From Json    
     final httpRequestData = json[OfflineDataKeys.httpRequestData.name]
    as List<DartIOHttpRequestData>;
return OfflineNetworkData(
      httpRequestData: httpRequestData,
      selectedRequestId:
      json[OfflineDataKeys.selectedRequestId.name] as String?,
      socketData: socketReqData,
    )
```
This still works, as dart is preserving those object types (when we don't serialise them, hence no need for deserialisation).
Since this offline feature is only used in dart and an in-memory thing, is this is good idea to not fully serialise & de-serialise it? and store the inner objects of DartIOHttpRequestData list and Socket and as they are and pick them? Or is it not recommended or safe to use?

Reply:
In order to download the screen's data as a JSON file, the serialization is required. However, not serializing the data is acceptable for the 'review history' feature, which allows continuing to view recent data offline when the connected app disconnects. I wouldn't worry about distinguishing the two for this PR. This can be a clean up later if the serialization / deserialization for the 'review history' features causes performance issues.

Reply:
Yeah sure, I was asking for 'review history' itself.